src/frontend/src/components/EditWizard/EditWizard.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+    Children,
+    cloneElement,
+    isValidElement,
+    useEffect,
+    useState,
+    type ReactNode,
+} from "react";
+import { useTranslation } from "react-i18next";
+import { useSelector } from "react-redux";
+
+import Wizard from "components/Wizard/Wizard";
+import { updateUserProfile } from "reducers/userProfile/userProfileThunk";
+import { useAppDispatch, type RootState } from "store/store";
+
+import _ from "lodash";
+import type { BaseProps } from "../../pages/Discover/Profile/types";
+
+type EditWizardChildComponent<P = {}> = React.ReactElement<P & BaseProps>;
+
+interface EditWizardProps {
+    isWizardOpen: boolean;
+    setWizardOpen: (isOpen: boolean) => void;
+    children: ReactNode;
+}
+
+const EditWizard = ({
+    isWizardOpen,
+    setWizardOpen,
+    children,
+}: EditWizardProps) => {
+    const profile = useSelector((state: RootState) => state.userProfile);
+    const [profileBuffer, setProfileBuffer] = useState({ ...profile });
+    const [currentStep, setCurrentStep] = useState(1);
+
+    const totalSteps = Children.count(children);
+
+    const dispatch = useAppDispatch();
+    const { t } = useTranslation();
+
+    // listen to profile change in redux store and update local buffer
+    useEffect(() => {
+        setProfileBuffer({ ...profile });
+    }, [profile]);
+
+    const handleClose = () => {
+        // clear buffers and states
+        setProfileBuffer({ ...profile });
+        setWizardOpen(false);
+        // Reset to first step when the modal is closed
+        setTimeout(() => setCurrentStep(1), 200);
+    };
+
+    // next button callback, only for navigation
+    const handleNext = () => {
+        if (currentStep < totalSteps) {
+            setCurrentStep(prev => prev + 1);
+        }
+    };
+
+    // previous button callback, only for navigation
+    const handleBack = () => {
+        if (currentStep > 1) {
+            setCurrentStep(prev => prev - 1);
+        }
+    };
+
+    // persist data to backend via PUT endpoint
+    // redux store's user profile gets updated on fulfilled PUT in the reducer
+    const handleFinish = () => {
+        const destinations = profileBuffer.destinations;
+        // One destination must be primary, used to set
+        // userProfile.activeDestination to calculate score.
+        // Use first destination by default.
+        if (!destinations.find(d => d.primary)) {
+            destinations[0] = { ...destinations[0], primary: true };
+        }
+        if (!_.isEqual(profileBuffer, profile)) {
```

Comment:
I like this, to avoid excessive API calls and unnecessary re-calculations when there's nothing changed.